### PR TITLE
gameflow: add stats screen and fade out to Lara's Home

### LINKED ISSUE #383: Add fade effect upon gym completion
@Richard-L can you split adding a fade effect to the gym level end into a separate issue?

_Originally posted by @rr- in https://github.com/rr-/Tomb1Main/issues/379#issuecomment-1026332914_

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #383.

#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
Added a fade out and stats screen at the end of Lara's Home.
...

### CODE REVIEW

**Comment by rr-** [comment]
That's good and all but let's make the stats screen exit early if the level is a gym.
(e.g. not show the stats since they  all have 0-0-0-0)

**Comment by walkawayy** [comment]
Fixed. Added an early gym check, fade, and return. Fades out without needing input too which matches the PS1 version.

File: src/game/stats.c [hunk 1]
```diff
@@ -259,6 +259,24 @@ int32_t Stats_GetSecrets(void)
...
+        Text_Draw();
+        Output_DumpScreen();
+        Output_FadeToBlack(false);
+        Text_RemoveAll();
+        while (Output_FadeIsAnimating()) {
+            Output_InitialisePolyList();
+            Game_DrawScene(false);
+            Output_DumpScreen();
+        }
+        Output_FadeReset();
```
Comment (rr-) [inline src/game/stats.c]: This should be able to be simplified to

```c
        Output_FadeToBlack(false);
        while (Output_FadeIsAnimating()) {
            Output_InitialisePolyList();
            Game_DrawScene(false);
            Output_DumpScreen();
        }
        Output_FadeReset();
```

Comment (walkawayy) [reply src/game/stats.c]: Thanks. Works great. I still don't understand the fade code fully. :D

**Review by rr- — APPROVED** [review]

**Pull request merged**